Fix #176 - Initial Pelias Geocoder using pelias-client-library dependency

jmfield2 opened the pull request:
Fix #176 and #337 

Note that not all results have the neighbourhood property set.

jmfield2 committed:
Fix #176 - Initial Pelias Geocoder using pelias-cutr-library dependency

barbeau commented on opentripplanner-geocoder/pom.xml:
```diff
@@ -121,6 +134,12 @@
+    </dependency>    
+    <!-- Pelias Client Library -->
+    <dependency>
+      <groupId>edu.cutr.pelias</groupId>    
```
After my fix for the Pelias client library in https://github.com/CUTR-at-USF/pelias-client-library/pull/10, this should now be:

```
<groupId>edu.usf.cutr.pelias</groupId>  
```

barbeau commented on opentripplanner-geocoder/src/main/resources/org/opentripplanner/geocoder/application-context.xml:
```diff
@@ -410,6 +410,13 @@
 	</bean>
 
+	<bean id="geocoderPelias" class="org.opentripplanner.geocoder.pelias.PeliasGeocoder">
+                <property name="apiKey" value="search-hAyPn56" />		
```
From https://github.com/CUTR-at-USF/usf-mobullity/issues/176#issuecomment-242398716, I believe this should be:

```
value="search-zso6JBE"
```

barbeau reviewed (changes requested):
Thanks @jmfield2!  Two small changes, and other than that looks good.

jmfield2 committed:
Update pom.xml package name and version, add geojson dependency
Change PeliasRequest to SearchRequest
Fix incorrect API key
Make Pelias the default instead of Nominatum

barbeau commented on opentripplanner-geocoder/pom.xml:
```diff
@@ -137,9 +142,9 @@
+      <groupId>edu.usf.cutr.pelias</groupId>    
       <artifactId>pelias-client-library</artifactId>    
-      <version>1.0.0-SNAPSHOT</version>
+      <version>1.0.1-SNAPSHOT</version>
```
As we discussed at meeting, this should be `1.0.0-SNAPSHOT`.

barbeau reviewed (changes requested):
Looks good, modulo artifact name!

barbeau commented:
Sorry, not artifact name, artifact version.

jmfield2 committed:
Update pelias artifactVersion to 1.0.0

barbeau commented on opentripplanner-geocoder/pom.xml:
```diff
@@ -31,6 +31,18 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
 
+<!-- CUTR SNAPSHOTs/RELEASES -->
```
You can actually go ahead and remove this `<repositories>` block - I finally got the v1.0.0 artifacts published to Maven Central last night, so we can use those.

barbeau commented on opentripplanner-geocoder/pom.xml:
```diff
@@ -121,6 +139,12 @@
+    <dependency>
+      <groupId>edu.usf.cutr</groupId>    
+      <artifactId>pelias-client-library</artifactId>    
+      <version>1.0.0-SNAPSHOT</version>
```
This can now change to `<version>1.0.0</version>`.

barbeau commented on opentripplanner-geocoder/src/main/java/org/opentripplanner/geocoder/pelias/PeliasGeocoder.java:
```diff
@@ -0,0 +1,167 @@
+                    .setSources("osm")
+                    .setSize(resultLimit)
+                    .setFocusPoint(focusLat, focusLon)
+                    .setBoundaryRect(rectMinLat, rectMinLon, rectMaxLat, rectMaxLon)                    
```
In v1.0.0, the `setBoundaryRect()` method now takes the `Double` datatype, instead of a string.  So you no longer need to do the `Double.toString()` operations above.

barbeau reviewed (changes requested):
@jmfield2 I was able to finally push a v1.0.0 release to Maven Central last night - some notes inline for switching to that release instead of v1.0.0-SNAPSHOT.

jmfield2 committed:
- Remove pelias-client-library local repository since it is in mavencentral
- Use 1.0.0 version instead of snapshot
- Change viewBox to an Envelope directly (with accompanying change in bean context.xml) to match what the pelias-client-library expects

jmfield2 commented:
The changes you requested should be good now.   I also changed the local viewBox to an Envelope (since it was pulling a string from the application-context.xml bean property ("x1,y1;x2,y2").

Now it will initialize the Envelope directly (using bean class= and constructor-arg) .. If you know of a better way to do that, let me know - otherwise, it seems to all work the same as before.

barbeau commented:
Thanks @jmfield2, looks good to me!
